dashboards-observability/public/components/visualizations/charts/maps/heatmap.tsx

```diff
@@ -37,8 +37,9 @@ export const HeatMap = ({ visualizations, layout, config }: any) => {
     isEmpty(zMetrics) ||
     isEmpty(data[xaxisField.label]) ||
     isEmpty(data[yaxisField.label]) ||
-    isEmpty(data[`${zMetrics.aggregation}(${zMetrics.name})`])
-    // indexOf(NUMERICAL_FIELDS, zMetrics.type) < 0
+    isEmpty(data[`${zMetrics.aggregation}(${zMetrics.name})`]) ||
+    dataConfig?.dimensions.length > 2 ||
```

Comment:
here do we need to add timestamp specific check?

Reply:
If we do not add timestamp in heatmap it still renders if the metric and dimensions are proper. So, we do not mandatorily need timestamp check.